<a href="https://colab.research.google.com/github/jose-gomezm/hacking-civico/blob/master/tareas/02_python_101.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Tarea 2

## Densidad poblacional

De acuerdo con la información otorgada por la [INEGI](http://cuentame.inegi.org.mx/monografias/informacion/yuc/poblacion/default.aspx?tema), cuya última actualización fue en 2015, la población para mi ciudad, estado y país son:

*   **Umán:** 55 261 habitantes
*   **Yucatán:** 2 097 175 habitantes
*   **México:** 119 530 753 habitantes

Por otra parte, las superficies de cada uno de ellos son:

* **Umán:** 349.56 km<sup>2</sup>
* **Yucatán:** 39 540.5 km<sup>2</sup>
* **México:** 1 959 248 km<sup>2</sup>

Para encontrar la densidad de población aplicamos la fórmula:

<a href="https://www.codecogs.com/eqnedit.php?latex=\dpi{120}&space;Densidad&space;=&space;\frac{Poblacion}{Superficie}" target="_blank"><img src="https://latex.codecogs.com/png.latex?\dpi{120}&space;Densidad&space;=&space;\frac{Poblacion}{Superficie}" title="Densidad = \frac{Poblacion}{Superficie}" /></a>



In [ ]:
def densidad(poblacion, superficie):
  return poblacion / superficie

Entonces para calcular la densidad de México, Yucatán, y Umán, definimos las variables con los valores de población y superficie para cada uno de ellos. Luego, usamos la función densidad

In [ ]:
pob_mexico = 119530753
sup_mexico = 1959248
pob_yucatan = 2097175
sup_yucatan = 39540.5
pob_uman = 55261
sup_uman = 349.56

den_mexico = densidad(pob_mexico, sup_mexico)
den_yucatan = densidad(pob_yucatan, sup_yucatan)
den_uman = densidad(pob_uman, sup_uman)

str_den = "Mexico: {:.4} hab/km2\nYucatan: {:.4} hab/km2\nUman: {:.5} hab/km2"

print(str_den.format(den_mexico, den_yucatan, den_uman))

Mexico: 61.01 hab/km2
Yucatan: 53.04 hab/km2
Uman: 158.09 hab/km2


**¿Cuál es la densidad poblacional?**

* México: 61.01 hab/km<sup>2</sup>
* Yucatán: 53.04 hab/km<sup>2</sup>
* Umán: 158.09 hab/km<sup>2</sup>

**¿Qué pasa comparar varios estados o ciudades?**

Podemos observar la diferencia entre medir una ciudad, un estado y un país. La densidad poblacional del estado de Yucatán y el país no difiere mucho en comparación a la de Umán. Probablemente la densidad poblacional de Mérida, la capital del estado, sea todavía mayor. Es de esperarse que la capital del estado presente una alta densidad, y que en comparación el estado sea menor. Gran parte de Yucatán es rural, con municipios pequeños y de pocos habitantes, lo que hace bajar la densidad en general como estado.

# Gasto público

De acuerdo con el sitio [Transparencia Presupuestaria](https://www.transparenciapresupuestaria.gob.mx/), a nivel federal en 2019 se destinó un total de 746,742.8 millones de pesos en el sector educativo, mientras que para 2018 se destinó 737,720.1 millones de pesos.

Para encontrar el gasto per cápita es necesario dividir el total de gasto entre el número de habitantes. Para el número de habitantes se utiliza el valor estimado, debido a que el valor de población es contado cada 5 años por la INEGI.

In [2]:
def per_capita(gasto, poblacion):
  return gasto / poblacion

In [3]:
ed_2018 = 737720100000
ed_2019 = 746742800000
pob_2018 = 124700000
pob_2019 = 	125930000

ed_pc_18 = per_capita(ed_2018, pob_2018)
ed_pc_19 = per_capita(ed_2019, pob_2019)

print("2018: {:.6}\n2019: {:.6}".format(ed_pc_18, ed_pc_19))

2018: 5915.96
2019: 5929.82


Para calcular el porcentaje, definimos la siguiente función:

In [4]:
def porcentaje(a, b):
  return (b - a) / b * 100

In [6]:
cambio = porcentaje(ed_pc_18, ed_pc_19)

print("Hubo un aumento del {:.2}%".format(cambio))

Hubo un aumento del 0.23%


**¿En qué porcentaje aumentó o disminuyó?**

Aumentó un 0.23%

**¿A qué crees que se deba?**

El porcentaje de aumento fue mínimo. El Gobierno federal ya había anunciado recortes presupuestarios al sector educativo, por lo que este valor puede ser el reflejo de ello. 

# Preguntas finales

**¿Para qué crees que nos pueda ser útil esta información?**

Como ciudadanos es importante conocer hacia dónde es dirigido el dinero destinado al gasto público. Esto permite desde verificar que las promesas de campaña sean cumplidas, hasta conocer las prioridades del gobierno en turno. 

**¿Qué decisiones podría tomar el gobierno basado en ésta?**

Debido a la condición actual que estamos pasando, el gobierno podría usar la información del gasto para contener los efectos que la contingencia podría tener en la economía.